astropy/config/paths.py

```diff
@@ -71,21 +71,31 @@ def _find_home():
             homedir = os.environ['HOME']
         else:
             raise OSError('Could not find a home directory to search for '
-                          'astropy config dir - are you on an unspported '
+                          'astropy config dir - are you on an unsupported '
                           'platform?')
     return homedir
 
 
-def get_config_dir(create=True):
+def get_config_dir(rootname, create=False):
     """
-    Determines the Astropy configuration directory name and creates the
+    Determines the package configuration directory name and creates the
     directory if it doesn't exist.
 
     This directory is typically ``$HOME/.astropy/config``, but if the
     XDG_CONFIG_HOME environment variable is set and the
     ``$XDG_CONFIG_HOME/astropy`` directory exists, it will be that directory.
     If neither exists, the former will be created and symlinked to the latter.
 
+    Parameters
+    ----------
+    rootname : str
+        Name of the root configuration directory. For example, if
+        ``rootname = 'pkgname'``, the configuration directory will be
+        ``<home>/.pkgname/``.
```

Comment:
Hmm, wasn't there a PR somewhere else trying to get us to move to more standard locations? Though arguably that's orthogonal to what is being done here.

Reply:
There is an issue #8245 I plan to open a PR for that after this one and post the 3.2 feature freeze.